server/tso/tso.go

```diff
@@ -195,7 +196,9 @@ func (t *timestampOracle) ResetUserTimestamp(leadership *election.Leadership, ts
 // 2. The physical time is monotonically increasing.
 // 3. The physical time is always less than the saved timestamp.
 func (t *timestampOracle) UpdateTimestamp(leadership *election.Leadership) error {
-	prev := (*atomicObject)(atomic.LoadPointer(&t.tso))
+	t.mux.Lock()
+	defer t.mux.Unlock()
```

Comment:
Cannot lock the whole function because it may too long to access etcd.
Should lock twice here. First a read lock to load, then save etcd, finally acquire a write lock to update.